generators/kubernetes/templates/console/_jhipster-elasticsearch.yml

```diff
@@ -16,41 +16,474 @@
  See the License for the specific language governing permissions and
  limitations under the License.
 -%>
-apiVersion: extensions/v1beta1
-kind: Deployment
+apiVersion: v1
+kind: ConfigMap
 metadata:
-  name: jhipster-elasticsearch
+  name: es-config
+  namespace: <%= kubernetesNamespace %>
+data:
+    discovery.type: zen
+    minimum.master.nodes: "1"
+    unicast.host: jhipster-elasticsearch-discovery
+    network.host: 0.0.0.0
+    http.enabled: "true"
+    memory.opts: -Xms512m -Xmx512m
+    cluster.name: jhipster-elasticsearch-cluster
+    memory.lock: "false"
+    path.data: /es/data
+    path.logs: /es/logs
+---
+apiVersion: apps/v1beta1
+kind: StatefulSet
+metadata:
+  name: jhipster-elasticsearch-master
   namespace: <%= kubernetesNamespace %>
+  labels:
+    component: jhipster-elasticsearch
+    role: master
 spec:
+  serviceName: jhipster-elasticsearch-master
   replicas: 1
   template:
     metadata:
       labels:
-        app: jhipster-elasticsearch
+        component: jhipster-elasticsearch
+        role: master
     spec:
+      initContainers:
+      - name: init-sysctl
+        image: busybox
+        # This would be needed to give access to 'elasticsearch' user that comes from the base
+        # image where the provisioned volume is host path (in minikube)
+        command:
+        - "/bin/sh"
+        - "-c"
+        - "chown -R 1000:1000 /es"
+        - sysctl
+        - -w
+        - vm.max_map_count=262144
+        securityContext:
+          runAsUser: 0
+        volumeMounts:
+        - name: storage
+          mountPath: /es
       containers:
-      - image: <%= DOCKER_JHIPSTER_ELASTICSEARCH %>
-        name: jhipster-elasticsearch
+      - name: jhipster-elasticsearch-master
+        image: <%= DOCKER_JHIPSTER_ELASTICSEARCH %>
+        env:
+        - name: NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: node.name
```

Comment:
This will break applications that are using kubernetes version less than 1.8

Reply:
Yes, supported in 1.8 and above

Reply:
Interesting do we have this documented somewhere. I mean that we support only 1.8 and above? 

Since by default the kubernetes client (atleast in europe zone) is v1.7.11 so this throws an error.

Reply:
yep, document needs to be updated. I thought about it. As it involves changing the config properties this is the way to go as es limits env prop by specifying the directory path instead of a specific file.